firebase-sessions/src/main/kotlin/com/google/firebase/sessions/settings/RemoteSettings.kt

```diff
@@ -81,6 +78,7 @@ internal class RemoteSettings(
 
     fetchInProgress.set(true)
 
+    // TODO(mrober): Avoid sending the fid here, and avoid fetching it when data collection is off.
```

Comment:
Ok so I looked in the iOS SDK - the way we handle this is just not kicking off the fetch unless we start sending events. So like where we log the event to FireLog is where we'd also kick off settings fetch (but settings must happen before we check the sampling rate)

In the iOS SDK there's a separate method for fetching settings and reading from the cache. Reading from the cache always happens, but fetching happens only when we're cleared to send events

Reply:
I did something very similar in https://github.com/firebase/firebase-android-sdk/pull/5107